app/controllers/interest_controller.rb

```diff
@@ -25,100 +25,133 @@ class InterestController < ApplicationController
   def list_interests
     store_location
     @title = :list_interests_title.t
-    # notifications = Notification.find_all_by_user_id(@user.id).sort do |a,b|
-    notifications = Notification.where(user_id: @user.id).sort do |a, b|
+    notifications = find_relevant_notifications
+    interests = find_relevant_interests
+    @targets = notifications + interests
+
+    @pages = paginate_numbers(:page, 50)
+    @pages.num_total = @targets.length
+    @targets = @targets[@pages.from..@pages.to]
+  end
+
+  private
+
+  def find_relevant_notifications
+    Notification.for_user(@user).sort do |a, b|
       result = a.flavor.to_s <=> b.flavor.to_s
       result = a.summary.to_s <=> b.summary.to_s if result.zero?
       result
     end
-    # interests = Interest.find_all_by_user_id(@user.id).sort do |a,b|
-    interests = Interest.where(user_id: @user.id).sort do |a, b|
+  end
+
+  def find_relevant_interests
+    Interest.for_user(@user).sort do |a, b|
       result = a.target_type <=> b.target_type
       if result.zero?
         result = (a.target ? a.target.text_name : "") <=>
                  (b.target ? b.target.text_name : "")
       end
       result
     end
-    @targets = notifications + interests
-    @pages = paginate_numbers(:page, 50)
-    @pages.num_total = @targets.length
-    @targets = @targets[@pages.from..@pages.to]
   end
 
+  public
+
   # Callback to change interest state in an object.
   # Linked from: show_<object> and emails
   # Redirects back (falls back on show_<object>)
   # Inputs: params[:type], params[:id], params[:state], params[:user]
   # Outputs: none
-  def set_interest
+  def set_interest # rubocop:disable Metrics/AbcSize
```

Comment:
While I personally have absolutely no problem with this method as-is... in this case it _could_ be debigulated in a few ways.

1. pass_query_params and requires_login could be outsourced to before_filters along the lines of other controllers.
2. you could move the find_object and error checking to another method, then the only params[:xxx] you need in this top level method is state, I think
3. even state could be moved out if you moved set_interest_state out into a fairly trivial submethod, too.

I can't believe rubocop has brainwashed me into arguing its case for it...  ¡Fight the system!

Reply:
Oops, I see you already do #3, ignore that one.